compiler/rustc_middle/src/ty/layout.rs

```diff
@@ -2776,9 +2768,9 @@ where
                     // LLVM's definition of `noalias` is based solely on memory
                     // dependencies rather than pointer equality
                     let no_alias = match kind {
-                        PointerKind::Shared => false,
+                        PointerKind::Shared | PointerKind::UniqueBorrowed => false,
```

Comment:
Why does `UniqueBorrowed` lead to `no_alias == false` now?

Oh I see, this is explained in `compiler/rustc_target/src/abi/call/mod.rs`. Maybe also add a similar comment here for people just reading this file?

Reply:
Thanks. :)